lib/board.rb

```diff
@@ -1,113 +1,101 @@
 class Board
-  attr_accessor :grid 
 
   def initialize(symbols = Array.new(9))
     @grid = symbols
   end
 
   def is_empty
-    !(grid.include?(:X) or grid.include?(:O))
+    !(grid.include?(PlayerSymbols::X) or grid.include?(PlayerSymbols::O))
   end
 
-  def update(index, symbol)
-    grid_before_index = copy_portion_of_grid(index)
-    grid_after_index = copy_portion_of_grid(grid.size - (index + 1))  
+  def make_move(index, symbol)
+    grid_before_index = copy_portion_of_grid(0, index)
+    grid_after_index = copy_portion_of_grid(index + 1, grid.size)   
 
-    set_grid Array.new(grid_before_index + [symbol] + grid_after_index)
+    Board.new(Array.new(grid_before_index + [symbol] + grid_after_index))
   end
 
   def has_free_spaces 
     grid.include?(nil)
   end
 
-  def has_winning_combination
-    all_winning_combinations = all_winning_combinations_for_3x3_grid
+  def get_symbol_at(position)
+    grid.at(position)
+  end
 
-    has_win_for_x = check_for_winning_row_of(:X, all_winning_combinations) 
-    has_win_for_o = check_for_winning_row_of(:O, all_winning_combinations)
+  def has_winning_combination
+    has_winning_combination_within(all_rows)
+  end
 
-    has_win_for_x or has_win_for_o
+  def winning_symbol
+    winning_row = find_winning_row_from(all_rows)
+    winning_row.nil? ? nil : winning_row.first 
   end
 
-  def check_for_winning_row_of(symbol, all_winning_combinations) 
-    all_indices_of_symbol = grid.each_index.select { |v| grid[v] == symbol} 
-    check_for_winning_combination_in(all_indices_of_symbol, all_winning_combinations)
+  def grid_for_display
+    rows
   end
 
-  def winning_symbol
-    all_winning_combinations = all_winning_combinations_for_3x3_grid
-    if check_for_winning_row_of(:X, all_winning_combinations) 
-      return :X 
-    end
+  private
 
-    if check_for_winning_row_of(:O, all_winning_combinations) 
-      return :O
-    end
+  attr_reader :grid
 
-    :unset
+  def copy_portion_of_grid(start_index, end_index)
+    grid[start_index...end_index] 
   end
 
-  def grid_for_display
-    grid_formation = ""
-    (0...grid.size).each do |i|
-      if grid[i].nil?
-        grid_formation+="empty$"
-      else 
-        grid_formation+= grid[i].to_s + "$"
-      end 
-    end
-    grid_formation 
+  def has_winning_combination_within(all_rows) 
+    has_row_of_matching_symbols_within(all_rows) 
   end
 
-  private
+  def has_row_of_matching_symbols_within(all_rows)
+    has_winning_row(all_rows) ? true : false
+  end
+
+  def has_winning_row(all_rows)
```

Comment:
Similar to above, suffix predicate with a ?. The name of this method can stay the same.